### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Navijava**

### CODE REVIEW

File: src/main/java/core/mate/academy/service/TruckProducer.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Truck;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
+    public List<Truck> get() {
+        Truck truck1 = new Truck("owner1", 1);
+        Truck truck2 = new Truck("owner1", 2);
+        Truck truck3 = new Truck("owner1", 3);
```
Comment (liliia-ponomarenko) [inline src/main/java/core/mate/academy/service/TruckProducer.java:10]: You can create objects using loop

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 2]
```diff
@@ -1,7 +1,42 @@
...
+            return new BulldozerProducer().get();
+        }
+        if (type == Excavator.class) {
+            return new ExcavatorProducer().get();
+        }
+        if (type == Truck.class) {
+            return new TruckProducer().get();
+        }
+        return new ArrayList<>();
+    }
```
Comment (liliia-ponomarenko) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:27]: you can use else-if else - else construction

File: src/main/java/core/mate/academy/model/Truck.java [hunk 3]
```diff
@@ -5,9 +5,21 @@
...
     }
 
+    public Truck(String owner, int volume) {
+        this.owner = owner;
+        this.volume = volume;
+    }
+
+    String getOwner() {
+        return owner;
+    }
```
Comment (liliia-ponomarenko) [inline src/main/java/core/mate/academy/model/Truck.java:21]: Let's create getters and setters for all fields and in all classes

**Review by liliia-ponomarenko — APPROVED** [review]

File: src/main/java/core/mate/academy/service/BulldozerProducer.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+import core.mate.academy.model.Bulldozer;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Bulldozer> {
+    public List<Bulldozer> get() {
+        Bulldozer bull1 = new Bulldozer("type1", 1);
+        Bulldozer bull2 = new Bulldozer("type2", 2);
+        Bulldozer bull3 = new Bulldozer("type3", 3);
+        return List.of(bull1, bull2, bull3);
```
Comment (KostyaNikiforov) [inline src/main/java/core/mate/academy/service/BulldozerProducer.java:11]:
```suggestion
        return List.of(new Bulldozer("type1", 1), new Bulldozer("type2", 2), new Bulldozer("type3", 3));
```
A matter of taste, up to you

File: src/main/java/core/mate/academy/service/TruckProducer.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+public class TruckProducer implements MachineProducer<Truck> {
+    public List<Truck> get() {
+        List<Truck> list = new ArrayList<>();
+        for (int i = 0; i < 3; i++) {
+            /*
+            Truck truck1 = new Truck("owner1", 1);
+            Truck truck2 = new Truck("owner1", 2);
+            Truck truck3 = new Truck("owner1", 3);
+            */
```
Comment (KostyaNikiforov) [inline src/main/java/core/mate/academy/service/TruckProducer.java:15]: Better remove redundant comments

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 6]
```diff
@@ -1,7 +1,42 @@
...
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+
```
Comment (KostyaNikiforov) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:16]: Empty line

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 7]
```diff
@@ -1,7 +1,42 @@
...
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> list = new ArrayList<>();
+        if (Bulldozer.class.equals(type)) {
+            list = new BulldozerProducer().get();
+        } else if (Excavator.class.equals(type)) {
+            list = new ExcavatorProducer().get();
+        } else if (Truck.class.equals(type)) {
+            list = new TruckProducer().get();
+        }
+        return (List<Machine>) list;
```
Comment (KostyaNikiforov) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:26]: And also we can use hashMap to make it prettier and more readable

**Review by KostyaNikiforov — CHANGES_REQUESTED** [review]

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java (hunk 7 above)
Comment (Navijava) [reply src/main/java/core/mate/academy/service/MachineServiceImpl.java]: Все поправил?

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java (hunk 2 above)
Comment (KostyaNikiforov) [reply src/main/java/core/mate/academy/service/MachineServiceImpl.java:25]: If you return something in the `if` block there is no need to use `else` in the next block

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 8]
```diff
@@ -1,7 +1,40 @@
...
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List getAll(Class type) {
+        if (type == Bulldozer.class) {
+            return new BulldozerProducer().get();
```
Comment (KostyaNikiforov) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:18]: You can avoid casting this way
```suggestion
            return new ArrayList<>(new BulldozerProducer().get());
```

**Review by KostyaNikiforov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**